pkg/nsx/services/subnet/subnet.go

```diff
@@ -418,3 +427,54 @@ func (service *SubnetService) UpdateSubnetSetTags(ns string, vpcSubnets []*model
 	}
 	return nil
 }
+
+func (service *SubnetService) validateSubnetImmutableFields(existingNsxSubnet *model.VpcSubnet, nsxSubnet *model.VpcSubnet) error {
+	errorMsg := ""
+
+	if *existingNsxSubnet.AccessMode != *nsxSubnet.AccessMode {
+		errorMsg = fmt.Sprintf("Subnet CR AccessMode cannot be modified")
+		return nsxutil.ImmutableFieldModifyError{Desc: errorMsg}
+	}
+
+	if *existingNsxSubnet.DhcpConfig.EnableDhcp != *nsxSubnet.DhcpConfig.EnableDhcp {
+		errorMsg = fmt.Sprintf("Subnet CRDHCPConfig cannot be modified")
+		return nsxutil.ImmutableFieldModifyError{Desc: errorMsg}
+	}
+
+	if existingNsxSubnet.Ipv4SubnetSize != nil && nsxSubnet.Ipv4SubnetSize != nil {
+		if *existingNsxSubnet.Ipv4SubnetSize != *nsxSubnet.Ipv4SubnetSize {
+			errorMsg = fmt.Sprintf("Subnet CR Ipv4SubnetSize cannot be modified")
```

Comment:
I think we shouldn't use the common logic in NSX operator to check the error. Consider the following case: the customer changed the subnet size from A to B, then the NSX operator updates the error on the status; however, the customer unluckily forgets the original correct size value A, then it seems that the subnet can never recover?
So it looks that we'd better use the webhook to do the check?